### DISCUSSION TITLE
画了一个 Web3:// + 区块链作为后端服务 框架示意图，不知道对不对？

### DISCUSSION THREAD

**Discussion by luffythink** [IntensiveCoLearning/Web3-URL#171] [Show and tell]
![](https://github.com/IntensiveCoLearning/Web3-URL/raw/main/img/web3serve.png)

理解有三个部分：
- Web3:// 如何从智能合约读取数据，可能涉及 ENS、Gateway、RPC
-  Web page 通过 Gateway / RPC 或是原生 EVM 浏览器呈现，所涉及的部分调用文件可以通过 EthStorage 存储服务支持
-  目前通过钱包连接 Web page 可实现与智能合约交互

不知道对不对，大家讨论一下。

**Comment by luffythink** [comment]
补充更新 EthStorage 部分：

![EthStorageworks](https://github.com/user-attachments/assets/91aa659a-d7d1-4c64-af1a-ad846a6967ff)

**Comment by qizhou** [comment]
非常棒！

其中有一小点在图里面注意的是，Gateway + RPC也可以通过原生浏览器 + Portal Network来去中心化访问(原文后面文字也有说明)。

**Reply by luffythink** [reply]
Get，看 Portal Network 的介绍是：由多个点对点网络组成，这些网络共同提供必要的数据和功能，以支持标准的 JSON-RPC API。这些网络特别设计主要目的是以确保 clients 在使用这些网络时能最小化「**网络带宽、CPU、RAM 和硬盘资源**」的消耗。🤔
